arcaflow_plugin_horreum_client/horreum_client_plugin.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+
+import sys
+import typing
+import requests
+import responses
+import json
+
+from arcaflow_plugin_sdk import plugin
+from horreum_client_schema import InputParams, SuccessOutput, ErrorOutput
+
+
+@plugin.step(
+    id="horreum-client",
+    name="Horreum Client",
+    description="Uploads an object to the Horreum server",
+    outputs={"success": SuccessOutput, "error": ErrorOutput},
+)
+def horreum_client(
+    params: InputParams,
+) -> typing.Tuple[str, typing.Union[SuccessOutput, ErrorOutput]]:
+    auth_url = (
+        params.horreum_keycloak_url + "/realms/horreum/protocol/openid-connect/token"
+    )
+    auth_obj = {
+        "username": params.horreum_username,
+        "password": params.horreum_password,
+        "grant_type": "password",
+        "client_id": "horreum-ui",
+    }
+
+    try:
+        print("==> Authenticating with keycloak...")
+        if params.mock:
+            with responses.RequestsMock() as rsps:
+                rsps.add(
+                    responses.POST,
+                    auth_url,
+                    json={"access_token": "mock_token"},
+                    status=200,
+                    content_type="application/json",
+                )
+                auth_return = requests.post(auth_url, data=auth_obj, verify=False)
+                assert auth_return.status_code == 200
+        else:
+            auth_return = requests.post(auth_url, data=auth_obj, verify=False)
+
+        token = json.loads(auth_return.text)["access_token"]
+
+    except requests.ConnectionError or requests.HTTPError or requests.Timeout:
```

Comment:
I'm not sure that the `or` operator is going to do what you want it to do, here.  (That is, I'm fairly sure that this `except` clause will catch _only_ `ConnectionError` and that it will _not_ catch the other two.)

The syntax that I think you're looking for is an iterable:
```python
    except (requests.ConnectionError, requests.HTTPError, requests.Timeout):
```
(And, of course, my comments apply below as well as above.)